1.segment_images/scripts/5.create_cytoplasm_masks.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[ ]:
+
+
+import argparse
+import pathlib
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import skimage.io as io
+import tifffile
+
+# check if in a jupyter notebook
+try:
+    cfg = get_ipython().config
+    in_notebook = True
+except NameError:
+    in_notebook = False
+
+
+# In[ ]:
+
+
+if not in_notebook:
+    print("Running as script")
+    # set up arg parser
+    parser = argparse.ArgumentParser(description="Segment the nuclei of a tiff image")
+
+    parser.add_argument(
+        "--input_dir",
+        type=str,
+        help="Path to the input directory containing the tiff images",
+    )
+
+    args = parser.parse_args()
+    input_dir = pathlib.Path(args.input_dir).resolve(strict=True)
+    mask_input_dir = pathlib.Path(f"../processed_data/{input_dir.stem}").resolve(
+        strict=True
+    )
+else:
+    print("Running in a notebook")
+    input_dir = pathlib.Path("../../data/z-stack_images/raw_z_input/").resolve(
+        strict=True
+    )
+    mask_input_dir = pathlib.Path(f"../processed_data/{input_dir.stem}").resolve(
+        strict=True
+    )
+
+
+output_path = pathlib.Path(f"../processed_data/{input_dir.stem}").resolve()
+output_path.mkdir(parents=True, exist_ok=True)
+output_file_path = pathlib.Path(output_path / "cytoplasm_mask.tiff").resolve()
+
+
+# In[3]:
+
+
+# get all the masks
+nuclei_masks_path = pathlib.Path(mask_input_dir / "nuclei_masks.tiff").resolve(
+    strict=True
+)
+cell_masks_path = pathlib.Path(mask_input_dir / "cell_masks.tiff").resolve(strict=True)
+
+nuclei_masks = io.imread(nuclei_masks_path)
+cell_masks = io.imread(cell_masks_path)
+
+
+# In[4]:
+
+
+cytoplasm_masks = np.zeros_like(cell_masks)
+for z_slice_index in range(nuclei_masks.shape[0]):
+    nuclei_slice_mask = nuclei_masks[z_slice_index]
+    cell_slice_mask = cell_masks[z_slice_index]
+    cytoplasm_mask = cell_slice_mask.copy()
+    cytoplasm_mask[nuclei_slice_mask > 0] = 0
+    cytoplasm_masks[z_slice_index] = cytoplasm_mask
+
```

Comment:
Consider adding a single comment documentation about what this for loop is doing